command-expansion-server/src/packages/lib/CommandTypeCodegen.ts

```diff
@@ -174,7 +174,17 @@ ${doc}
 function mapArgumentType(argument: ampcs.FswCommandArgument, enumMap: ampcs.EnumMap): string {
   switch (argument.arg_type) {
     case "enum":
-      return enumMap[argument.enum_name].values.map((value) => `'${value.symbol}'`).join(" | ");
+      // boolean enum shouldn't be "TRUE | FALSE" but of `boolean` type
+      if (
+        enumMap[argument.enum_name].values.length === 2 &&
+        enumMap[argument.enum_name].values.filter(({ symbol, numeric }) => {
```

Comment:
100% stylistic and nit-picky 

For readability, can we consider switching this to a 
```
enumMap[argument.enum_name].values.length === 2
&& enumMap[argument.enum_name].values.some(({ symbol, numeric }) => symbol.toLocaleLowerCase() === 'true')
&& enumMap[argument.enum_name].values.some(({ symbol, numeric }) => symbol.toLocaleLowerCase() === 'false')
```

So it reads as "Must have 2 variants AND one must be true AND one must be false?

Approved for merge regardless

Reply:
I just re-read this as "Must have 2 variants and there must be none that are not true and are not false"

So I'm going to retract my previous comment.

Reply:
An enum with the values "true" and "TrUe" would match the code as written I think.

So I just want to point out that there is a semantic difference between Dylan's suggestion and the code as is... does it matter? 🤷

Reply:
Is there a particular reason we want this match to be case insensitive?

Reply:
If the enum is of length 2 and the users add 'TrUe', then most likely intent is the user is probably meant 'true'.

Reply:
These values aren't coming from users though, right? They're coming from merlin server:

https://github.com/NASA-AMMOS/aerie/blob/ad11d0997b86a04f27efacc1212197f6fe3252ef/merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/SerializedValueJsonParser.java#L67-L69

Now I suppose you could consider the developers of merlin-server "users" since it's from a different component... I'm not sure what the best way to go is. I'm not going to request changes, but I want to call out that there is an edge case that would cause wrong behavior: namely an enum that has "true" in it twice, with different capitalizations.

Reply:
Good point on the double true matching though Matt. I think we want the incantation of `"Must have 2 variants AND one must be true AND one must be false"`

Reply:
Re. the capitalization, yeah as Ryan said, it's pretty unambiguous that they intend a boolean if they have 'TrUe' and 'fALse', despite them acting like a sea sponge

Reply:
@mattdailis the values are coming from the user's command_dictionary file.

Reply:
I can make the change to `"Must have 2 variants AND one must be true AND one must be false"`